### ISSUE TITLE
Unable to use System.Threading.Tasks when Firebase is imported

### ISSUE THREAD

**Issue by dbidwell94** [firebase/quickstart-unity#621]
### Please fill in the following fields:
Unity editor version: 

2020.1.0b3.3.3385

Firebase Unity SDK version:

 6.13.0

Firebase plugins in use (Auth, Database, etc.):

Auth

Additional SDKs you are using (Facebook, AdMob, etc.): 

None

Platform you are using the Unity editor on (Mac, Windows, or Linux): 

Windows

Platform you are targeting (iOS, Android, and/or desktop): 

Windows, Linux, Mac, Android

Scripting Runtime (Mono, and/or IL2CPP):

IL2CPP (Android)
Mono (Desktop)

### Please describe the issue here:
(Please list the full steps to reproduce the issue. Include device logs, Unity logs, and stack traces if available.)

Open blank Unity project. Import any Firebase unitypackage (Auth was used for me). Create a new C# script. Import Firebase, FirebaseAuth, and System.Threading.Tasks. Create a custom method using async Task<T> and an error will occur saying that the namespace System.Threading.Tasks is used in both Unity.Tasks and mscorelib

### Please answer the following, if applicable:
Have you been able to reproduce this issue with just the Firebase Unity quickstarts (this GitHub project)?

Yes, upon creating a new C# script, added a System.Threading.Tasks using statement, and creating a custom async method that returns a Task<T>

What's the issue repro rate? (eg 100%, 1/5 etc)

100%

`
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using System.IO;
using System;
using System.Threading.Tasks;
using Firebase;

public static class SomeGenericClassName
{
    private static async Task<LevelSaveData> GetLevelSaveAsync(string fileName)
    {
        LevelSave save = await Task.Run(() => LoadLevelFromDrive(fileName));
        LevelSaveData data = new LevelSaveData();
        data.levelName = save.levelName;
        Texture2D shotTex = new Texture2D(2, 2);
        shotTex.LoadImage(save.rawPicData);
        shotTex.Apply();
        Sprite screenshot = Sprite.Create(shotTex, new Rect(0, 0, shotTex.width, shotTex.height), new Vector2());
        data.screenshot = screenshot;
        data.tex = shotTex;
        data.levelSavePath = GetSavePath(data.levelName);
        return data;
    }
}
`

Above is a highly modified version of the script that I am using that throws the error. However, Unity will still compile the project and the game will still run. It _might_ be an issue with Visual Studio. I believe the issue is because Firebase ships with Unity.Tasks dll which also contains the System.Threading.Tasks namespace. I do not believe this is necessary anymore due to the fact that Unity by default now has access to System.Threading.Tasks namespace. Unity.Tasks.dll is an unnecessary redundancy but unfortunately if you delete it from the project it will break the Firebase plugin.

**Comment by dbidwell94** [comment]
![image](https://user-images.githubusercontent.com/25497078/77811956-534b4c80-705b-11ea-8292-7803ce434ed5.png)

**Comment by dbidwell94** [comment]
Is there not a way to tell firebase to use System.Threading.Tasks from mscorelib instead of bundling it's own in a 3rd party assembly that interferes with the rest of my project's code?

**Comment by alexames** [comment]
Have a look at [this page](https://firebase.google.com/docs/unity/setup#net_compatibility_when_using_unity_2017x_and_later) for information on what to do. It sounds like you may have installed the wrong .unitypackages for the version of .NET you're using. Can you verify that you're using the right version and follow the instructions on that page and see if that resolves your issue?

**Comment by chkuang-g** [comment]
Hi @dbidwell94 
Since you are using the latest and newest Unity 2020, it only support .NET 4.x and .NET Standard 2.0.

That is, when you import Firebase, please select those `.unitypackage` under **dotnot4** folder.

TL;DR; Those under `dotnet3` folders have Task class implemented by Parse since .NET 3.x simply does not Task at all.

Hope this helps!

**Comment by dbidwell94** [comment]
Resolved by importing via package manager instead of the *.unitypackage
